src/main/main.ts

```diff
@@ -0,0 +1,70 @@
+import { PROJECT_ID } from "./basecamp";
+import { generateIdForRow, getBasecampTodoRequestsForRow, getId, hasId, saveRow } from "./row";
+import { getEventRowsFromSpreadsheet } from "./scan";
+import { createTodo, TODOLIST_ID } from "./todos";
+
+const DEFAULT_TODOLIST_IDENTIFIER: TodolistIdentifier = {
+    projectId: PROJECT_ID,
+    todolistId: TODOLIST_ID
+};
+
+/**
+ * Main entry point for the Onestop to Basecamp Integration that contains the core logic for
+ * adding/updating/deleting Basecamp Todos based on the rows on the Onestop
+ */
+export function main(): void {
+    const eventRows: Row[] = getEventRowsFromSpreadsheet();
+    const processedRowIds: string[] = [];
+
+    for(const eventRow of eventRows) {
+        if(hasId(eventRow)) {
+            processExistingRow(eventRow);
+        } else {
+            processNewRow(eventRow);
+        }
+
+        processedRowIds.push(getId(eventRow));
+    }
+
+    deleteOldRows(processedRowIds);
+}
+
+function processExistingRow(row: Row): void {
+    // Will be implemented as part of https://3.basecamp.com/4474129/buckets/38736474/todos/7717428992
+}
+
+/**
+ * Processes new event rows by creating a new Basecamp Todo for each of the roles associated with that 
+ * particular event row. If at least on Todo is creaed in Bascamp, the row is saved for easier retrieval
+ * and updating later
+ * 
+ * @param row 
+ */
+function processNewRow(row: Row): void {
+    const basecampTodoRequests: BasecampTodoRequest[] = getBasecampTodoRequestsForRow(row);
+    const basecampTodoIds: string[] = createNewTodos(basecampTodoRequests);
+
+    if(basecampTodoIds.length > 0) {
```

Comment:
in the case of an execution crash, to me it feels safer to immediately save the row id once the basecamp request succeeds, what do you think? otherwise worst case you have a bunch of orphaned todos in basecamp

Reply:
@andrewc5716 Just for clarification, each row can have multiple Basecamp Todo Requests. Are you saying to save the row after each individual request? 

Another option is we could also wrap the Http request call in a try catch to catch and handle any errors so execution doesn't halt and we can continuing processing as normal saving rows with successful basecamp todo ids so they don't get orphaned. Ideally we would also have some retry mechanism/exponential backoff

Another thing that comes to mind is if one row has 3 Basecamp Todo Requests and request 1 and 2 are successful but request 3 fails. If we continue processing and save the row with ids for request 1 and 2, we have no way of knowing in the future that we need to come back and retry 3 at a later time. We may need a mechanism to handle this